### PR TITLE
GC: Restore: SharePoint: Page Logic

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Restore logic for restoring a SharePoint Page to M365 given a valid `[]byte`.  
Delete API also included

Tests included

## Does this PR need a docs update or release note?
- [x] :no_entry: No 

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* related to  #2169<issue>

## Test Plan

- [x] :zap: Unit test
Must be tested locally due to CI Library issues, See #2086.
Clean-up is handled within the tests.

### CODE REVIEW

**Review by ryanfkeepers — APPROVED** [review]

File: src/internal/connector/sharepoint/restore.go [hunk 1]
```diff
@@ -270,3 +270,68 @@ func RestoreCollection(
...
+		return dii, fmt.Errorf("created page %s did not return webURL from API", *restoredPage.GetId())
+	}
+
+	pageID := *restoredPage.GetId()
+
+	err = service.Client().SitesById(siteID).PagesById(pageID).Publish().Post(ctx, nil)
+	if err != nil {
+		return dii, support.ConnectorStackErrorTraceWrap(
+			err,
+			"unable to publish page ID: "+pageID+" API Error Details",
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/sharepoint/restore.go:335]: Resolved with [e77fd9c](https://github.com/alcionai/corso/pull/2225/commits/e77fd9c0d81c877ff027714277ed1357e9ae4e27)

File: src/internal/connector/sharepoint/restore.go [hunk 2]
```diff
@@ -270,3 +270,68 @@ func RestoreCollection(
...
+	defer end()
+
+	var (
+		dii      = details.ItemInfo{}
+		pageName = itemData.UUID()
+	)
+
+	byteArray, err := io.ReadAll(itemData.ToReader())
+	if err != nil {
+		return dii, errors.Wrap(err, "sharepoint restorePage failed to restore bytes from data.Stream")
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/sharepoint/restore.go:292]: Resolved with [e77fd9c](https://github.com/alcionai/corso/pull/2225/commits/e77fd9c0d81c877ff027714277ed1357e9ae4e27)

File: src/internal/connector/sharepoint/restore.go [hunk 3]
```diff
@@ -270,3 +270,68 @@ func RestoreCollection(
...
+	}
+
+	// Hydrate Page
+	page, err := support.CreatePageFromBytes(byteArray)
+	if err != nil {
+		return dii, errors.Wrapf(err, "failed to create Page object %s", pageName)
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/sharepoint/restore.go]: Resolved with [e77fd9c](https://github.com/alcionai/corso/pull/2225/commits/e77fd9c0d81c877ff027714277ed1357e9ae4e27)

File: src/internal/connector/sharepoint/restore.go [hunk 4]
```diff
@@ -270,3 +270,68 @@ func RestoreCollection(
...
+	fmt.Printf("Page Name: %s\n", *page.GetName())
+	// Restore is a 2-Step Process in Graph API
+	// 1. Create the Page on the site
+	// 2. Publish the site
+	// See: https://learn.microsoft.com/en-us/graph/api/sitepage-create?view=graph-rest-beta
+	restoredPage, err := service.Client().SitesById(siteID).Pages().Post(ctx, page, nil)
+	if err != nil {
+		sendErr := support.ConnectorStackErrorTraceWrap(
+			err,
+			"failure to create page from ID: %s"+pageName+" API Error Details",
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/sharepoint/restore.go:317]: Resolved with [e77fd9c](https://github.com/alcionai/corso/pull/2225/commits/e77fd9c0d81c877ff027714277ed1357e9ae4e27)

File: src/internal/connector/sharepoint/restore.go [hunk 5]
```diff
@@ -270,3 +270,68 @@ func RestoreCollection(
...
+
+	// Hydrate Page
+	page, err := support.CreatePageFromBytes(byteArray)
+	if err != nil {
+		return dii, errors.Wrapf(err, "failed to create Page object %s", pageName)
+	}
+
+	newName := fmt.Sprintf("%s_%s", destName, *page.GetName())
+	page.SetName(&newName)
+	fmt.Printf("Page Name: %s\n", *page.GetName())
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/sharepoint/restore.go]: Resolved with [e77fd9c](https://github.com/alcionai/corso/pull/2225/commits/e77fd9c0d81c877ff027714277ed1357e9ae4e27)

File: src/internal/connector/sharepoint/restore.go [hunk 6]
```diff
@@ -270,3 +270,68 @@ func RestoreCollection(
...
+	service graph.Servicer,
+	itemData data.Stream,
+	siteID, destName string,
+) (details.ItemInfo, error) {
+	ctx, end := D.Span(ctx, "gc:sharepoint:restorePage", D.Label("item_uuid", itemData.UUID()))
+	defer end()
+
+	var (
+		dii      = details.ItemInfo{}
+		pageName = itemData.UUID()
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/sharepoint/restore.go]: Resolved with [e77fd9c](https://github.com/alcionai/corso/pull/2225/commits/e77fd9c0d81c877ff027714277ed1357e9ae4e27)

File: src/internal/connector/sharepoint/collection_test.go [hunk 7]
```diff
@@ -161,6 +161,54 @@ func (suite *SharePointCollectionSuite) TestRestoreListCollection() {
...
+
+	service, err := createTestService(account)
+	require.NoError(t, err)
+
+	destName := "Corso_Restore_" + common.FormatNow(common.SimpleTimeTesting)
+	testName := "MockPage"
+
+	// Create Test Page
+	//nolint:lll
+	byteArray := []byte("{\"name\":\"test.aspx\",\"title\":\"test\",\"pageLayout\":\"article\",\"showComments\":true," +
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/sharepoint/collection_test.go:179]: Can we address this in https://github.com/alcionai/corso/pull/2226? 
We can address the formatting there.

**Pull request merged**